src/main/java/hudson/plugins/jira/JiraRestService.java

```diff
@@ -300,11 +300,22 @@ public BasicIssue createIssue(String projectKey, String description, String assi
             builder.setComponentsNames(components);
         }
 
-        final IssueInput issueInput = builder.build();
+        IssueInput issueInput = builder.build();
```

Comment:
fyi I already looked into detecting what type of Jira installation it is, there is an endpoint but the field isn't available in the dto.. /rest/api/2/serverInfo "deploymentType":"Server"